cypress/e2e/main.cy.js

```diff
@@ -30,14 +28,70 @@ describe('Main Page', () => {
 
     cy.get('[data-cy="movie-poster"] img').last().should('have.attr', 'alt', 'Pulp Fiction poster');
     cy.get('[data-cy="vote-container"]').last().should("exist")
-
-    cy.get('[data-cy="upvote-button"]').first().click()
-    // cy.get(".voteCount").should("contain", onIncreaseVote)
   })
 
-  // it('Increment the vote count')
+  it('updates vote count with up and down buttons', () => {
+  cy.get(".voteCount").first().should("contain", 32544)
+  cy.get('[data-cy="upvote-button"]').first().click()
+
+  
+  cy.get(".voteCount").first().should("contain", 32545)
+  cy.intercept("PATCH", "https://rancid-tomatillos-api-cc6f59111a05.herokuapp.com/api/v1/movies/155", {
+    statusCode: 200,
+    fixture: "movie_posters"
+  })
+  cy.visit('http://localhost:3000/')
+  
+  cy.get(".voteCount").first().should("contain", 32544)
+  cy.get('[data-cy="downvote-button"]').first().click()
 
+  
+  cy.get(".voteCount").first().should("contain", 32543)
+  cy.intercept("PATCH", "https://rancid-tomatillos-api-cc6f59111a05.herokuapp.com/api/v1/movies/155", {
+    statusCode: 200,
+    fixture: "movie_posters"
+  })
+  cy.visit('http://localhost:3000/')
+  
+  })
 
+  it('displays an error message if the vote update fails', () => {
+    cy.intercept("PATCH", "https://rancid-tomatillos-api-cc6f59111a05.herokuapp.com/api/v1/movies/155", {
+      statusCode: 500,
+      body: { error: 'Internal Server Error' }
+    });
+      cy.get('[data-cy="upvote-button"]').first().click();
+  
+      cy.get('.error-message').should('contain', 'Failed to update vote. Please try again later.');
+  });
+  it('displays an error message when movies fail to load', () => {
```

Comment:
Good to know we have some edge casing, cannot imagine a circumstance when it would be triggered, but we will be prepared when it does.